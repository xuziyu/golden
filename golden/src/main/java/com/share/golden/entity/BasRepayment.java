package com.share.golden.entity;

import java.util.Date;

public class BasRepayment {
    private String repayId;

    private String repayCode;

    private Date repayDate;

    private String repayState;

    private String repayTime;

    private String repayRemark1;

    private String repayRemark2;

    public String getRepayId() {
        return repayId;
    }

    public void setRepayId(String repayId) {
        this.repayId = repayId == null ? null : repayId.trim();
    }

    public String getRepayCode() {
        return repayCode;
    }

    public void setRepayCode(String repayCode) {
        this.repayCode = repayCode == null ? null : repayCode.trim();
    }

    public Date getRepayDate() {
        return repayDate;
    }

    public void setRepayDate(Date repayDate) {
        this.repayDate = repayDate;
    }

    public String getRepayState() {
        return repayState;
    }

    public void setRepayState(String repayState) {
        this.repayState = repayState == null ? null : repayState.trim();
    }

    public String getRepayTime() {
        return repayTime;
    }

    public void setRepayTime(String repayTime) {
        this.repayTime = repayTime == null ? null : repayTime.trim();
    }

    public String getRepayRemark1() {
        return repayRemark1;
    }

    public void setRepayRemark1(String repayRemark1) {
        this.repayRemark1 = repayRemark1 == null ? null : repayRemark1.trim();
    }

    public String getRepayRemark2() {
        return repayRemark2;
    }

    public void setRepayRemark2(String repayRemark2) {
        this.repayRemark2 = repayRemark2 == null ? null : repayRemark2.trim();
    }
}