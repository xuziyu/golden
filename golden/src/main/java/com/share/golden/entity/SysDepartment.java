package com.share.golden.entity;

import java.util.Date;

public class SysDepartment {
    private String departmentId;

    private String parentDepartmentId;

    private String departmentName;

    private Date departmentTime;

    private String departmentStatus;

    private String departmentEscribes;

    public String getDepartmentId() {
        return departmentId;
    }

    public void setDepartmentId(String departmentId) {
        this.departmentId = departmentId == null ? null : departmentId.trim();
    }

    public String getParentDepartmentId() {
        return parentDepartmentId;
    }

    public void setParentDepartmentId(String parentDepartmentId) {
        this.parentDepartmentId = parentDepartmentId == null ? null : parentDepartmentId.trim();
    }

    public String getDepartmentName() {
        return departmentName;
    }

    public void setDepartmentName(String departmentName) {
        this.departmentName = departmentName == null ? null : departmentName.trim();
    }

    public Date getDepartmentTime() {
        return departmentTime;
    }

    public void setDepartmentTime(Date departmentTime) {
        this.departmentTime = departmentTime;
    }

    public String getDepartmentStatus() {
        return departmentStatus;
    }

    public void setDepartmentStatus(String departmentStatus) {
        this.departmentStatus = departmentStatus == null ? null : departmentStatus.trim();
    }

    public String getDepartmentEscribes() {
        return departmentEscribes;
    }

    public void setDepartmentEscribes(String departmentEscribes) {
        this.departmentEscribes = departmentEscribes == null ? null : departmentEscribes.trim();
    }
}