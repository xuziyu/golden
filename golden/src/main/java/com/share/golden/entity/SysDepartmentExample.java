package com.share.golden.entity;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class SysDepartmentExample {
    protected String orderByClause;

    protected boolean distinct;

    protected List<Criteria> oredCriteria;

    public SysDepartmentExample() {
        oredCriteria = new ArrayList<Criteria>();
    }

    public void setOrderByClause(String orderByClause) {
        this.orderByClause = orderByClause;
    }

    public String getOrderByClause() {
        return orderByClause;
    }

    public void setDistinct(boolean distinct) {
        this.distinct = distinct;
    }

    public boolean isDistinct() {
        return distinct;
    }

    public List<Criteria> getOredCriteria() {
        return oredCriteria;
    }

    public void or(Criteria criteria) {
        oredCriteria.add(criteria);
    }

    public Criteria or() {
        Criteria criteria = createCriteriaInternal();
        oredCriteria.add(criteria);
        return criteria;
    }

    public Criteria createCriteria() {
        Criteria criteria = createCriteriaInternal();
        if (oredCriteria.size() == 0) {
            oredCriteria.add(criteria);
        }
        return criteria;
    }

    protected Criteria createCriteriaInternal() {
        Criteria criteria = new Criteria();
        return criteria;
    }

    public void clear() {
        oredCriteria.clear();
        orderByClause = null;
        distinct = false;
    }

    protected abstract static class GeneratedCriteria {
        protected List<Criterion> criteria;

        protected GeneratedCriteria() {
            super();
            criteria = new ArrayList<Criterion>();
        }

        public boolean isValid() {
            return criteria.size() > 0;
        }

        public List<Criterion> getAllCriteria() {
            return criteria;
        }

        public List<Criterion> getCriteria() {
            return criteria;
        }

        protected void addCriterion(String condition) {
            if (condition == null) {
                throw new RuntimeException("Value for condition cannot be null");
            }
            criteria.add(new Criterion(condition));
        }

        protected void addCriterion(String condition, Object value, String property) {
            if (value == null) {
                throw new RuntimeException("Value for " + property + " cannot be null");
            }
            criteria.add(new Criterion(condition, value));
        }

        protected void addCriterion(String condition, Object value1, Object value2, String property) {
            if (value1 == null || value2 == null) {
                throw new RuntimeException("Between values for " + property + " cannot be null");
            }
            criteria.add(new Criterion(condition, value1, value2));
        }

        public Criteria andDepartmentIdIsNull() {
            addCriterion("\"DEPARTMENT_ID\" is null");
            return (Criteria) this;
        }

        public Criteria andDepartmentIdIsNotNull() {
            addCriterion("\"DEPARTMENT_ID\" is not null");
            return (Criteria) this;
        }

        public Criteria andDepartmentIdEqualTo(String value) {
            addCriterion("\"DEPARTMENT_ID\" =", value, "departmentId");
            return (Criteria) this;
        }

        public Criteria andDepartmentIdNotEqualTo(String value) {
            addCriterion("\"DEPARTMENT_ID\" <>", value, "departmentId");
            return (Criteria) this;
        }

        public Criteria andDepartmentIdGreaterThan(String value) {
            addCriterion("\"DEPARTMENT_ID\" >", value, "departmentId");
            return (Criteria) this;
        }

        public Criteria andDepartmentIdGreaterThanOrEqualTo(String value) {
            addCriterion("\"DEPARTMENT_ID\" >=", value, "departmentId");
            return (Criteria) this;
        }

        public Criteria andDepartmentIdLessThan(String value) {
            addCriterion("\"DEPARTMENT_ID\" <", value, "departmentId");
            return (Criteria) this;
        }

        public Criteria andDepartmentIdLessThanOrEqualTo(String value) {
            addCriterion("\"DEPARTMENT_ID\" <=", value, "departmentId");
            return (Criteria) this;
        }

        public Criteria andDepartmentIdLike(String value) {
            addCriterion("\"DEPARTMENT_ID\" like", value, "departmentId");
            return (Criteria) this;
        }

        public Criteria andDepartmentIdNotLike(String value) {
            addCriterion("\"DEPARTMENT_ID\" not like", value, "departmentId");
            return (Criteria) this;
        }

        public Criteria andDepartmentIdIn(List<String> values) {
            addCriterion("\"DEPARTMENT_ID\" in", values, "departmentId");
            return (Criteria) this;
        }

        public Criteria andDepartmentIdNotIn(List<String> values) {
            addCriterion("\"DEPARTMENT_ID\" not in", values, "departmentId");
            return (Criteria) this;
        }

        public Criteria andDepartmentIdBetween(String value1, String value2) {
            addCriterion("\"DEPARTMENT_ID\" between", value1, value2, "departmentId");
            return (Criteria) this;
        }

        public Criteria andDepartmentIdNotBetween(String value1, String value2) {
            addCriterion("\"DEPARTMENT_ID\" not between", value1, value2, "departmentId");
            return (Criteria) this;
        }

        public Criteria andParentDepartmentIdIsNull() {
            addCriterion("\"PARENT_DEPARTMENT_ID\" is null");
            return (Criteria) this;
        }

        public Criteria andParentDepartmentIdIsNotNull() {
            addCriterion("\"PARENT_DEPARTMENT_ID\" is not null");
            return (Criteria) this;
        }

        public Criteria andParentDepartmentIdEqualTo(String value) {
            addCriterion("\"PARENT_DEPARTMENT_ID\" =", value, "parentDepartmentId");
            return (Criteria) this;
        }

        public Criteria andParentDepartmentIdNotEqualTo(String value) {
            addCriterion("\"PARENT_DEPARTMENT_ID\" <>", value, "parentDepartmentId");
            return (Criteria) this;
        }

        public Criteria andParentDepartmentIdGreaterThan(String value) {
            addCriterion("\"PARENT_DEPARTMENT_ID\" >", value, "parentDepartmentId");
            return (Criteria) this;
        }

        public Criteria andParentDepartmentIdGreaterThanOrEqualTo(String value) {
            addCriterion("\"PARENT_DEPARTMENT_ID\" >=", value, "parentDepartmentId");
            return (Criteria) this;
        }

        public Criteria andParentDepartmentIdLessThan(String value) {
            addCriterion("\"PARENT_DEPARTMENT_ID\" <", value, "parentDepartmentId");
            return (Criteria) this;
        }

        public Criteria andParentDepartmentIdLessThanOrEqualTo(String value) {
            addCriterion("\"PARENT_DEPARTMENT_ID\" <=", value, "parentDepartmentId");
            return (Criteria) this;
        }

        public Criteria andParentDepartmentIdLike(String value) {
            addCriterion("\"PARENT_DEPARTMENT_ID\" like", value, "parentDepartmentId");
            return (Criteria) this;
        }

        public Criteria andParentDepartmentIdNotLike(String value) {
            addCriterion("\"PARENT_DEPARTMENT_ID\" not like", value, "parentDepartmentId");
            return (Criteria) this;
        }

        public Criteria andParentDepartmentIdIn(List<String> values) {
            addCriterion("\"PARENT_DEPARTMENT_ID\" in", values, "parentDepartmentId");
            return (Criteria) this;
        }

        public Criteria andParentDepartmentIdNotIn(List<String> values) {
            addCriterion("\"PARENT_DEPARTMENT_ID\" not in", values, "parentDepartmentId");
            return (Criteria) this;
        }

        public Criteria andParentDepartmentIdBetween(String value1, String value2) {
            addCriterion("\"PARENT_DEPARTMENT_ID\" between", value1, value2, "parentDepartmentId");
            return (Criteria) this;
        }

        public Criteria andParentDepartmentIdNotBetween(String value1, String value2) {
            addCriterion("\"PARENT_DEPARTMENT_ID\" not between", value1, value2, "parentDepartmentId");
            return (Criteria) this;
        }

        public Criteria andDepartmentNameIsNull() {
            addCriterion("\"DEPARTMENT_NAME\" is null");
            return (Criteria) this;
        }

        public Criteria andDepartmentNameIsNotNull() {
            addCriterion("\"DEPARTMENT_NAME\" is not null");
            return (Criteria) this;
        }

        public Criteria andDepartmentNameEqualTo(String value) {
            addCriterion("\"DEPARTMENT_NAME\" =", value, "departmentName");
            return (Criteria) this;
        }

        public Criteria andDepartmentNameNotEqualTo(String value) {
            addCriterion("\"DEPARTMENT_NAME\" <>", value, "departmentName");
            return (Criteria) this;
        }

        public Criteria andDepartmentNameGreaterThan(String value) {
            addCriterion("\"DEPARTMENT_NAME\" >", value, "departmentName");
            return (Criteria) this;
        }

        public Criteria andDepartmentNameGreaterThanOrEqualTo(String value) {
            addCriterion("\"DEPARTMENT_NAME\" >=", value, "departmentName");
            return (Criteria) this;
        }

        public Criteria andDepartmentNameLessThan(String value) {
            addCriterion("\"DEPARTMENT_NAME\" <", value, "departmentName");
            return (Criteria) this;
        }

        public Criteria andDepartmentNameLessThanOrEqualTo(String value) {
            addCriterion("\"DEPARTMENT_NAME\" <=", value, "departmentName");
            return (Criteria) this;
        }

        public Criteria andDepartmentNameLike(String value) {
            addCriterion("\"DEPARTMENT_NAME\" like", value, "departmentName");
            return (Criteria) this;
        }

        public Criteria andDepartmentNameNotLike(String value) {
            addCriterion("\"DEPARTMENT_NAME\" not like", value, "departmentName");
            return (Criteria) this;
        }

        public Criteria andDepartmentNameIn(List<String> values) {
            addCriterion("\"DEPARTMENT_NAME\" in", values, "departmentName");
            return (Criteria) this;
        }

        public Criteria andDepartmentNameNotIn(List<String> values) {
            addCriterion("\"DEPARTMENT_NAME\" not in", values, "departmentName");
            return (Criteria) this;
        }

        public Criteria andDepartmentNameBetween(String value1, String value2) {
            addCriterion("\"DEPARTMENT_NAME\" between", value1, value2, "departmentName");
            return (Criteria) this;
        }

        public Criteria andDepartmentNameNotBetween(String value1, String value2) {
            addCriterion("\"DEPARTMENT_NAME\" not between", value1, value2, "departmentName");
            return (Criteria) this;
        }

        public Criteria andDepartmentTimeIsNull() {
            addCriterion("\"DEPARTMENT_TIME\" is null");
            return (Criteria) this;
        }

        public Criteria andDepartmentTimeIsNotNull() {
            addCriterion("\"DEPARTMENT_TIME\" is not null");
            return (Criteria) this;
        }

        public Criteria andDepartmentTimeEqualTo(Date value) {
            addCriterion("\"DEPARTMENT_TIME\" =", value, "departmentTime");
            return (Criteria) this;
        }

        public Criteria andDepartmentTimeNotEqualTo(Date value) {
            addCriterion("\"DEPARTMENT_TIME\" <>", value, "departmentTime");
            return (Criteria) this;
        }

        public Criteria andDepartmentTimeGreaterThan(Date value) {
            addCriterion("\"DEPARTMENT_TIME\" >", value, "departmentTime");
            return (Criteria) this;
        }

        public Criteria andDepartmentTimeGreaterThanOrEqualTo(Date value) {
            addCriterion("\"DEPARTMENT_TIME\" >=", value, "departmentTime");
            return (Criteria) this;
        }

        public Criteria andDepartmentTimeLessThan(Date value) {
            addCriterion("\"DEPARTMENT_TIME\" <", value, "departmentTime");
            return (Criteria) this;
        }

        public Criteria andDepartmentTimeLessThanOrEqualTo(Date value) {
            addCriterion("\"DEPARTMENT_TIME\" <=", value, "departmentTime");
            return (Criteria) this;
        }

        public Criteria andDepartmentTimeIn(List<Date> values) {
            addCriterion("\"DEPARTMENT_TIME\" in", values, "departmentTime");
            return (Criteria) this;
        }

        public Criteria andDepartmentTimeNotIn(List<Date> values) {
            addCriterion("\"DEPARTMENT_TIME\" not in", values, "departmentTime");
            return (Criteria) this;
        }

        public Criteria andDepartmentTimeBetween(Date value1, Date value2) {
            addCriterion("\"DEPARTMENT_TIME\" between", value1, value2, "departmentTime");
            return (Criteria) this;
        }

        public Criteria andDepartmentTimeNotBetween(Date value1, Date value2) {
            addCriterion("\"DEPARTMENT_TIME\" not between", value1, value2, "departmentTime");
            return (Criteria) this;
        }

        public Criteria andDepartmentStatusIsNull() {
            addCriterion("\"DEPARTMENT_STATUS\" is null");
            return (Criteria) this;
        }

        public Criteria andDepartmentStatusIsNotNull() {
            addCriterion("\"DEPARTMENT_STATUS\" is not null");
            return (Criteria) this;
        }

        public Criteria andDepartmentStatusEqualTo(String value) {
            addCriterion("\"DEPARTMENT_STATUS\" =", value, "departmentStatus");
            return (Criteria) this;
        }

        public Criteria andDepartmentStatusNotEqualTo(String value) {
            addCriterion("\"DEPARTMENT_STATUS\" <>", value, "departmentStatus");
            return (Criteria) this;
        }

        public Criteria andDepartmentStatusGreaterThan(String value) {
            addCriterion("\"DEPARTMENT_STATUS\" >", value, "departmentStatus");
            return (Criteria) this;
        }

        public Criteria andDepartmentStatusGreaterThanOrEqualTo(String value) {
            addCriterion("\"DEPARTMENT_STATUS\" >=", value, "departmentStatus");
            return (Criteria) this;
        }

        public Criteria andDepartmentStatusLessThan(String value) {
            addCriterion("\"DEPARTMENT_STATUS\" <", value, "departmentStatus");
            return (Criteria) this;
        }

        public Criteria andDepartmentStatusLessThanOrEqualTo(String value) {
            addCriterion("\"DEPARTMENT_STATUS\" <=", value, "departmentStatus");
            return (Criteria) this;
        }

        public Criteria andDepartmentStatusLike(String value) {
            addCriterion("\"DEPARTMENT_STATUS\" like", value, "departmentStatus");
            return (Criteria) this;
        }

        public Criteria andDepartmentStatusNotLike(String value) {
            addCriterion("\"DEPARTMENT_STATUS\" not like", value, "departmentStatus");
            return (Criteria) this;
        }

        public Criteria andDepartmentStatusIn(List<String> values) {
            addCriterion("\"DEPARTMENT_STATUS\" in", values, "departmentStatus");
            return (Criteria) this;
        }

        public Criteria andDepartmentStatusNotIn(List<String> values) {
            addCriterion("\"DEPARTMENT_STATUS\" not in", values, "departmentStatus");
            return (Criteria) this;
        }

        public Criteria andDepartmentStatusBetween(String value1, String value2) {
            addCriterion("\"DEPARTMENT_STATUS\" between", value1, value2, "departmentStatus");
            return (Criteria) this;
        }

        public Criteria andDepartmentStatusNotBetween(String value1, String value2) {
            addCriterion("\"DEPARTMENT_STATUS\" not between", value1, value2, "departmentStatus");
            return (Criteria) this;
        }

        public Criteria andDepartmentEscribesIsNull() {
            addCriterion("\"DEPARTMENT_ESCRIBES\" is null");
            return (Criteria) this;
        }

        public Criteria andDepartmentEscribesIsNotNull() {
            addCriterion("\"DEPARTMENT_ESCRIBES\" is not null");
            return (Criteria) this;
        }

        public Criteria andDepartmentEscribesEqualTo(String value) {
            addCriterion("\"DEPARTMENT_ESCRIBES\" =", value, "departmentEscribes");
            return (Criteria) this;
        }

        public Criteria andDepartmentEscribesNotEqualTo(String value) {
            addCriterion("\"DEPARTMENT_ESCRIBES\" <>", value, "departmentEscribes");
            return (Criteria) this;
        }

        public Criteria andDepartmentEscribesGreaterThan(String value) {
            addCriterion("\"DEPARTMENT_ESCRIBES\" >", value, "departmentEscribes");
            return (Criteria) this;
        }

        public Criteria andDepartmentEscribesGreaterThanOrEqualTo(String value) {
            addCriterion("\"DEPARTMENT_ESCRIBES\" >=", value, "departmentEscribes");
            return (Criteria) this;
        }

        public Criteria andDepartmentEscribesLessThan(String value) {
            addCriterion("\"DEPARTMENT_ESCRIBES\" <", value, "departmentEscribes");
            return (Criteria) this;
        }

        public Criteria andDepartmentEscribesLessThanOrEqualTo(String value) {
            addCriterion("\"DEPARTMENT_ESCRIBES\" <=", value, "departmentEscribes");
            return (Criteria) this;
        }

        public Criteria andDepartmentEscribesLike(String value) {
            addCriterion("\"DEPARTMENT_ESCRIBES\" like", value, "departmentEscribes");
            return (Criteria) this;
        }

        public Criteria andDepartmentEscribesNotLike(String value) {
            addCriterion("\"DEPARTMENT_ESCRIBES\" not like", value, "departmentEscribes");
            return (Criteria) this;
        }

        public Criteria andDepartmentEscribesIn(List<String> values) {
            addCriterion("\"DEPARTMENT_ESCRIBES\" in", values, "departmentEscribes");
            return (Criteria) this;
        }

        public Criteria andDepartmentEscribesNotIn(List<String> values) {
            addCriterion("\"DEPARTMENT_ESCRIBES\" not in", values, "departmentEscribes");
            return (Criteria) this;
        }

        public Criteria andDepartmentEscribesBetween(String value1, String value2) {
            addCriterion("\"DEPARTMENT_ESCRIBES\" between", value1, value2, "departmentEscribes");
            return (Criteria) this;
        }

        public Criteria andDepartmentEscribesNotBetween(String value1, String value2) {
            addCriterion("\"DEPARTMENT_ESCRIBES\" not between", value1, value2, "departmentEscribes");
            return (Criteria) this;
        }
    }

    public static class Criteria extends GeneratedCriteria {

        protected Criteria() {
            super();
        }
    }

    public static class Criterion {
        private String condition;

        private Object value;

        private Object secondValue;

        private boolean noValue;

        private boolean singleValue;

        private boolean betweenValue;

        private boolean listValue;

        private String typeHandler;

        public String getCondition() {
            return condition;
        }

        public Object getValue() {
            return value;
        }

        public Object getSecondValue() {
            return secondValue;
        }

        public boolean isNoValue() {
            return noValue;
        }

        public boolean isSingleValue() {
            return singleValue;
        }

        public boolean isBetweenValue() {
            return betweenValue;
        }

        public boolean isListValue() {
            return listValue;
        }

        public String getTypeHandler() {
            return typeHandler;
        }

        protected Criterion(String condition) {
            super();
            this.condition = condition;
            this.typeHandler = null;
            this.noValue = true;
        }

        protected Criterion(String condition, Object value, String typeHandler) {
            super();
            this.condition = condition;
            this.value = value;
            this.typeHandler = typeHandler;
            if (value instanceof List<?>) {
                this.listValue = true;
            } else {
                this.singleValue = true;
            }
        }

        protected Criterion(String condition, Object value) {
            this(condition, value, null);
        }

        protected Criterion(String condition, Object value, Object secondValue, String typeHandler) {
            super();
            this.condition = condition;
            this.value = value;
            this.secondValue = secondValue;
            this.typeHandler = typeHandler;
            this.betweenValue = true;
        }

        protected Criterion(String condition, Object value, Object secondValue) {
            this(condition, value, secondValue, null);
        }
    }
}