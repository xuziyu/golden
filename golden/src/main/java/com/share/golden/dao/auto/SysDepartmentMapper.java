package com.share.golden.dao.auto;

import com.share.golden.entity.SysDepartment;
import com.share.golden.entity.SysDepartmentExample;
import java.util.List;
import org.apache.ibatis.annotations.Param;

public interface SysDepartmentMapper {
    long countByExample(SysDepartmentExample example);

    int deleteByExample(SysDepartmentExample example);

    int deleteByPrimaryKey(String departmentId);

    int insert(SysDepartment record);

    int insertSelective(SysDepartment record);

    List<SysDepartment> selectByExample(SysDepartmentExample example);

    SysDepartment selectByPrimaryKey(String departmentId);

    int updateByExampleSelective(@Param("record") SysDepartment record, @Param("example") SysDepartmentExample example);

    int updateByExample(@Param("record") SysDepartment record, @Param("example") SysDepartmentExample example);

    int updateByPrimaryKeySelective(SysDepartment record);

    int updateByPrimaryKey(SysDepartment record);
}