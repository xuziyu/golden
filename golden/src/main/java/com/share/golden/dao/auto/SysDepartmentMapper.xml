<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE mapper PUBLIC "-//mybatis.org//DTD Mapper 3.0//EN" "http://mybatis.org/dtd/mybatis-3-mapper.dtd">
<mapper namespace="com.share.golden.dao.auto.SysDepartmentMapper">
  <resultMap id="BaseResultMap" type="com.share.golden.entity.SysDepartment">
    <id column="DEPARTMENT_ID" jdbcType="VARCHAR" property="departmentId" />
    <result column="PARENT_DEPARTMENT_ID" jdbcType="VARCHAR" property="parentDepartmentId" />
    <result column="DEPARTMENT_NAME" jdbcType="VARCHAR" property="departmentName" />
    <result column="DEPARTMENT_TIME" jdbcType="TIMESTAMP" property="departmentTime" />
    <result column="DEPARTMENT_STATUS" jdbcType="VARCHAR" property="departmentStatus" />
    <result column="DEPARTMENT_ESCRIBES" jdbcType="VARCHAR" property="departmentEscribes" />
  </resultMap>
  <sql id="Example_Where_Clause">
    <where>
      <foreach collection="oredCriteria" item="criteria" separator="or">
        <if test="criteria.valid">
          <trim prefix="(" prefixOverrides="and" suffix=")">
            <foreach collection="criteria.criteria" item="criterion">
              <choose>
                <when test="criterion.noValue">
                  and ${criterion.condition}
                </when>
                <when test="criterion.singleValue">
                  and ${criterion.condition} #{criterion.value}
                </when>
                <when test="criterion.betweenValue">
                  and ${criterion.condition} #{criterion.value} and #{criterion.secondValue}
                </when>
                <when test="criterion.listValue">
                  and ${criterion.condition}
                  <foreach close=")" collection="criterion.value" item="listItem" open="(" separator=",">
                    #{listItem}
                  </foreach>
                </when>
              </choose>
            </foreach>
          </trim>
        </if>
      </foreach>
    </where>
  </sql>
  <sql id="Update_By_Example_Where_Clause">
    <where>
      <foreach collection="example.oredCriteria" item="criteria" separator="or">
        <if test="criteria.valid">
          <trim prefix="(" prefixOverrides="and" suffix=")">
            <foreach collection="criteria.criteria" item="criterion">
              <choose>
                <when test="criterion.noValue">
                  and ${criterion.condition}
                </when>
                <when test="criterion.singleValue">
                  and ${criterion.condition} #{criterion.value}
                </when>
                <when test="criterion.betweenValue">
                  and ${criterion.condition} #{criterion.value} and #{criterion.secondValue}
                </when>
                <when test="criterion.listValue">
                  and ${criterion.condition}
                  <foreach close=")" collection="criterion.value" item="listItem" open="(" separator=",">
                    #{listItem}
                  </foreach>
                </when>
              </choose>
            </foreach>
          </trim>
        </if>
      </foreach>
    </where>
  </sql>
  <sql id="Base_Column_List">
    "DEPARTMENT_ID", "PARENT_DEPARTMENT_ID", "DEPARTMENT_NAME", "DEPARTMENT_TIME", "DEPARTMENT_STATUS", 
    "DEPARTMENT_ESCRIBES"
  </sql>
  <select id="selectByExample" parameterType="com.share.golden.entity.SysDepartmentExample" resultMap="BaseResultMap">
    select
    <if test="distinct">
      distinct
    </if>
    <include refid="Base_Column_List" />
    from SYS_DEPARTMENT
    <if test="_parameter != null">
      <include refid="Example_Where_Clause" />
    </if>
    <if test="orderByClause != null">
      order by ${orderByClause}
    </if>
  </select>
  <select id="selectByPrimaryKey" parameterType="java.lang.String" resultMap="BaseResultMap">
    select 
    <include refid="Base_Column_List" />
    from SYS_DEPARTMENT
    where "DEPARTMENT_ID" = #{departmentId,jdbcType=VARCHAR}
  </select>
  <delete id="deleteByPrimaryKey" parameterType="java.lang.String">
    delete from SYS_DEPARTMENT
    where "DEPARTMENT_ID" = #{departmentId,jdbcType=VARCHAR}
  </delete>
  <delete id="deleteByExample" parameterType="com.share.golden.entity.SysDepartmentExample">
    delete from SYS_DEPARTMENT
    <if test="_parameter != null">
      <include refid="Example_Where_Clause" />
    </if>
  </delete>
  <insert id="insert" parameterType="com.share.golden.entity.SysDepartment">
    insert into SYS_DEPARTMENT ("DEPARTMENT_ID", "PARENT_DEPARTMENT_ID", 
      "DEPARTMENT_NAME", "DEPARTMENT_TIME", "DEPARTMENT_STATUS", 
      "DEPARTMENT_ESCRIBES")
    values (#{departmentId,jdbcType=VARCHAR}, #{parentDepartmentId,jdbcType=VARCHAR}, 
      #{departmentName,jdbcType=VARCHAR}, #{departmentTime,jdbcType=TIMESTAMP}, #{departmentStatus,jdbcType=VARCHAR}, 
      #{departmentEscribes,jdbcType=VARCHAR})
  </insert>
  <insert id="insertSelective" parameterType="com.share.golden.entity.SysDepartment">
    insert into SYS_DEPARTMENT
    <trim prefix="(" suffix=")" suffixOverrides=",">
      <if test="departmentId != null">
        "DEPARTMENT_ID",
      </if>
      <if test="parentDepartmentId != null">
        "PARENT_DEPARTMENT_ID",
      </if>
      <if test="departmentName != null">
        "DEPARTMENT_NAME",
      </if>
      <if test="departmentTime != null">
        "DEPARTMENT_TIME",
      </if>
      <if test="departmentStatus != null">
        "DEPARTMENT_STATUS",
      </if>
      <if test="departmentEscribes != null">
        "DEPARTMENT_ESCRIBES",
      </if>
    </trim>
    <trim prefix="values (" suffix=")" suffixOverrides=",">
      <if test="departmentId != null">
        #{departmentId,jdbcType=VARCHAR},
      </if>
      <if test="parentDepartmentId != null">
        #{parentDepartmentId,jdbcType=VARCHAR},
      </if>
      <if test="departmentName != null">
        #{departmentName,jdbcType=VARCHAR},
      </if>
      <if test="departmentTime != null">
        #{departmentTime,jdbcType=TIMESTAMP},
      </if>
      <if test="departmentStatus != null">
        #{departmentStatus,jdbcType=VARCHAR},
      </if>
      <if test="departmentEscribes != null">
        #{departmentEscribes,jdbcType=VARCHAR},
      </if>
    </trim>
  </insert>
  <select id="countByExample" parameterType="com.share.golden.entity.SysDepartmentExample" resultType="java.lang.Long">
    select count(*) from SYS_DEPARTMENT
    <if test="_parameter != null">
      <include refid="Example_Where_Clause" />
    </if>
  </select>
  <update id="updateByExampleSelective" parameterType="map">
    update SYS_DEPARTMENT
    <set>
      <if test="record.departmentId != null">
        "DEPARTMENT_ID" = #{record.departmentId,jdbcType=VARCHAR},
      </if>
      <if test="record.parentDepartmentId != null">
        "PARENT_DEPARTMENT_ID" = #{record.parentDepartmentId,jdbcType=VARCHAR},
      </if>
      <if test="record.departmentName != null">
        "DEPARTMENT_NAME" = #{record.departmentName,jdbcType=VARCHAR},
      </if>
      <if test="record.departmentTime != null">
        "DEPARTMENT_TIME" = #{record.departmentTime,jdbcType=TIMESTAMP},
      </if>
      <if test="record.departmentStatus != null">
        "DEPARTMENT_STATUS" = #{record.departmentStatus,jdbcType=VARCHAR},
      </if>
      <if test="record.departmentEscribes != null">
        "DEPARTMENT_ESCRIBES" = #{record.departmentEscribes,jdbcType=VARCHAR},
      </if>
    </set>
    <if test="_parameter != null">
      <include refid="Update_By_Example_Where_Clause" />
    </if>
  </update>
  <update id="updateByExample" parameterType="map">
    update SYS_DEPARTMENT
    set "DEPARTMENT_ID" = #{record.departmentId,jdbcType=VARCHAR},
      "PARENT_DEPARTMENT_ID" = #{record.parentDepartmentId,jdbcType=VARCHAR},
      "DEPARTMENT_NAME" = #{record.departmentName,jdbcType=VARCHAR},
      "DEPARTMENT_TIME" = #{record.departmentTime,jdbcType=TIMESTAMP},
      "DEPARTMENT_STATUS" = #{record.departmentStatus,jdbcType=VARCHAR},
      "DEPARTMENT_ESCRIBES" = #{record.departmentEscribes,jdbcType=VARCHAR}
    <if test="_parameter != null">
      <include refid="Update_By_Example_Where_Clause" />
    </if>
  </update>
  <update id="updateByPrimaryKeySelective" parameterType="com.share.golden.entity.SysDepartment">
    update SYS_DEPARTMENT
    <set>
      <if test="parentDepartmentId != null">
        "PARENT_DEPARTMENT_ID" = #{parentDepartmentId,jdbcType=VARCHAR},
      </if>
      <if test="departmentName != null">
        "DEPARTMENT_NAME" = #{departmentName,jdbcType=VARCHAR},
      </if>
      <if test="departmentTime != null">
        "DEPARTMENT_TIME" = #{departmentTime,jdbcType=TIMESTAMP},
      </if>
      <if test="departmentStatus != null">
        "DEPARTMENT_STATUS" = #{departmentStatus,jdbcType=VARCHAR},
      </if>
      <if test="departmentEscribes != null">
        "DEPARTMENT_ESCRIBES" = #{departmentEscribes,jdbcType=VARCHAR},
      </if>
    </set>
    where "DEPARTMENT_ID" = #{departmentId,jdbcType=VARCHAR}
  </update>
  <update id="updateByPrimaryKey" parameterType="com.share.golden.entity.SysDepartment">
    update SYS_DEPARTMENT
    set "PARENT_DEPARTMENT_ID" = #{parentDepartmentId,jdbcType=VARCHAR},
      "DEPARTMENT_NAME" = #{departmentName,jdbcType=VARCHAR},
      "DEPARTMENT_TIME" = #{departmentTime,jdbcType=TIMESTAMP},
      "DEPARTMENT_STATUS" = #{departmentStatus,jdbcType=VARCHAR},
      "DEPARTMENT_ESCRIBES" = #{departmentEscribes,jdbcType=VARCHAR}
    where "DEPARTMENT_ID" = #{departmentId,jdbcType=VARCHAR}
  </update>
</mapper>